game-core/src/main/java/games/strategy/util/ExitStatus.java

```diff
@@ -0,0 +1,27 @@
+package games.strategy.util;
+
+/**
+ * A process exit status.
+ */
+public enum ExitStatus {
+  /** The process exited successfully (0). */
+  SUCCESS(0),
+
+  /** The process exited due to a failure (1). */
+  FAILURE(1);
+
+  private final int status;
+
+  private ExitStatus(final int status) {
+    assert status >= 0 : "exit status must be non-negative to avoid signal aliasing";
+
+    this.status = status;
+  }
+
+  /**
+   * Exits the host process with this status.
+   */
+  public void exit() {
```

Comment:
Any chance we can mark this as deprecated?  : P

I'm half joking. To another extent `System.exit` does get flagged as non-portable. Going to single game instance as well it's more important to not exit unless we really are sure.

(no follow-up or response needed here, I'm half ribbing you @ssoloff and making a making a side comment)

Reply:
@DanVanAtta agreed.
It's really bad practice UI code or similar can just terminate the complete application, which makes multiple game instances per JVM instance impossible.